virtcontainers/qemu.go

```diff
@@ -748,10 +760,20 @@ func (q *qemu) qmpSetup() error {
 	}
 	q.qmpMonitorCh.qmp = qmp
 	q.qmpMonitorCh.disconn = disconnectCh
+	go q.loopQMPEvent(events)
 
 	return nil
 }
 
+func (q *qemu) loopQMPEvent(event chan govmmQemu.QMPEvent) {
```

Comment:
If the qemu is killed instead of exit by sending SHUTDOWN cmd from qmp, would this loopQMPEvent catch this case?

Reply:
@lifupan , yes, it can be catched

Reply:
Hi @Ace-Tang 
Sorry, I cannot catch when and where the qmp shutdown event was created when qemu was killed?

Reply:
:confused:   @lifupan , Can you describe how you test

Reply:
It's simple, run a pod and then kill the qemu process,  and I didn't see any shutdown events was logged.

Reply:
I am not test with k8s, but only run with ctr run, and kill the qemu, then I can get event
```
time="2019-05-15T00:25:02.069044069+08:00" level=info msg="{\"timestamp\": {\"seconds\": 1557851102, \"microseconds\": 68970}, \"event\": \"SHUTDOWN\", \"data\": {\"guest\": false}}" ID=kill-k1 source=virtcontainers subsystem=qmp
time="2019-05-15T00:25:02.069167555+08:00" level=info msg="log event SHUTDOWN" ID=kill-k1 source=virtcontainers subsystem=qemu
time="2019-05-15T00:25:02.069678935+08:00" level=error msg="Wait for process failed" container=kill-k1 error="rpc error: code = Unavailable desc = transport is closing" pid=kill-k1
```

Reply:
@Ace-Tang  can you also get when you do a `kill -9` ?

Reply:
No, so this method is not always work.